xarray/backends/api.py

```diff
@@ -1031,11 +1036,7 @@ def open_mfdataset(
             ds.close()
         raise
 
-    def multi_file_closer():
-        for closer in closers:
-            closer()
-
-    combined.set_close(multi_file_closer)
+    combined.set_close(partial(_multi_file_closer, closers))
```

Comment:
You'll need a `from functools import partial` at the top of this file too

Reply:
Handled via 13d82f4